flink-runtime/src/test/java/org/apache/flink/runtime/util/FlinkSecurityManagerITCase.java

```diff
@@ -0,0 +1,143 @@
+        } finally {
+            testProcess.destroy();
+        }
+    }
+
+    @Test
+    public void testIgnoredJVMExit() throws Exception {
+        final ForcedJVMExitProcess testProcess =
+                new ForcedJVMExitProcess(IgnoredExitEntryPoint.class);
+
+        try {
+            testProcess.startProcess();
+            testProcess.waitFor();
+            assertThat(testProcess.exitCode(), is(0));
+
+        } finally {
+            testProcess.destroy();
+        }
+    }
+
+    // ------------------------------------------------------------------------
+    //  Blocking Process Implementation
+    // ------------------------------------------------------------------------
+
+    private static final class ForcedJVMExitProcess extends TestJvmProcess {
+
+        private final Class<?> entryPointName;
+
+        private ForcedJVMExitProcess(Class<?> entryPointName) throws Exception {
+            this.entryPointName = entryPointName;
+        }
+
+        @Override
+        public String getName() {
+            return getEntryPointClassName();
+        }
+
+        @Override
+        public String[] getJvmArgs() {
+            return new String[0];
+        }
+
+        @Override
+        public String getEntryPointClassName() {
+            return entryPointName.getName();
+        }
+    }
+
+    // ------------------------------------------------------------------------
+
+    public static final class ForcedExitEntryPoint {
+
+        public static void main(String[] args) throws Exception {
+            Configuration configuration = new Configuration();
+            // configure FlinkSecurityManager to intercept calls to System.exit().
+            configuration.set(
+                    ClusterOptions.INTERCEPT_USER_SYSTEM_EXIT,
+                    ClusterOptions.UserSystemExitMode.THROW);
+            FlinkSecurityManager.setFromConfiguration(configuration);
+
+            FlinkSecurityManager.forceProcessExit(222);
+
+            System.exit(0);
+        }
+    }
+
+    public static final class IgnoredExitEntryPoint {
+
+        public static void main(String[] args) throws Exception {
+            Configuration configuration = new Configuration();
+            // configure FlinkSecurityManager to intercept calls to System.exit().
+            configuration.set(
+                    ClusterOptions.INTERCEPT_USER_SYSTEM_EXIT,
+                    ClusterOptions.UserSystemExitMode.THROW);
+            FlinkSecurityManager.setFromConfiguration(configuration);
+
+            FlinkSecurityManager.monitorUserSystemExitForCurrentThread();
+            // expect this call to be ignored
+            try {
+                System.exit(123);
+            } catch (Throwable t) {
+                System.err.println(
+                        "Caught exception during system exit with message: " + t.getMessage());
+            }
+
+            System.err.println("Test has passed");
```

Comment:
this can be misleading because it is also printed if System.exit threw an exception